x-pack/filebeat/module/virustotal/livehunt/_meta/fields.yml

```diff
@@ -0,0 +1,924 @@
+  fields:
+    - name: flattened
+      release: beta
+      type: group
+      fields:
+        - name: debug
+          default_field: false
+          description: >
+            Debug information, if present
+          type: flattened
+        - name: import_list
+          default_field: false
+          description: >
+            List of all imported functions
+          type: flattened
+        - name: sections
+          default_field: false
+          description: >
+            Data about sections of compiled binary PE
+          type: flattened
+        - name: resource_details
+          default_field: false
+          type: flattened
+          description: >
+            If the PE contains resources, some info about them
+        - name: resource_languages
+          default_field: false
+          type: flattened
+          description: >
+            Digest of languages found in resources. Key is language (as string) and
+            value is how many resources there are having that language (as integer)
+        - name: resource_types
+          default_field: false
+          type: flattened
+          description: >
+            Digest of resource types. Key is resource type (as string) and value
+            is how many resources there are of that specific type (as integer)
+        - name: packers
+          default_field: false
+          type: flattened
+          description: >
+            Identifies packers used on Windows PE files by several tools and AVs.
+            Keys are tool names and values are identified packers, both strings.
+            see `file.pe.packers` for merged list of packers from all tools.
+      overwrite: true
+    - name: exports
+      type: keyword
+      description: >
+        List of symbols exported by PE
+
+    - name: creation_date
+      default_field: false
+      description: >
+        extracted when possible from the file's metadata. Indicates when it was
+        built or compiled. It can also be faked by malware creators.
+      type: date
+    - name: authentihash
+      description: >
+        Authentihash of the PE file.
+      type: keyword
+    - name: compile_timestamp
+      description: >
+        Compile timestamp of the PE file.
+      type: date
+
+    - name: packers
+      type: keyword
+      description: >
+        Merged list of all detected packers by all tools used. See `file.pe.flattened.packers`.
+
+    - name: main_icon.hash
+      type: group
+      description: >
+        Hashes of embedded program icon
+      fields:
+        - name: dhash
+          description: >
+            Difference Hash for a given PE file.
+          type: keyword
+          release: beta
+        - name: md5
+          type: keyword
+          description: >
+            MD5 hash of raw icon data
+          release: beta
+    - name: compiler_product_versions
+      type: keyword
+      description: >
+        Version of the compiler product.
+    - name: rich_pe_header_hash
```

Comment:
does it make sense to make this into `rich_header.hash.*`? I would imagine that some other forensics from rich headers might be useful in other PE parsing implementations

Reply:
That's a good point. Since it's PE specific, maybe we treat it like `authentihash`. We could put them all under `file.pe.hash.*` with `authentihash`, `rich_header_hash`, `imphash`. Similarly, ELF would have `file.elf.hash.telfhash`

Reply:
I guess I was thinking more along the lines of making it possible for someone to actually namespace whatever parsing might be done on the rich header itself. Say, if someone wanted to try and actually parse out the [artifact ids/counts from the rich header itself](https://github.com/lief-project/LIEF/blob/master/src/PE/RichEntry.cpp#L29), then by doing something like `pe.rich_header.hash.*` you could allow for someone else to go in and do something like `pe.rich_header.entries` or something.

Additionally, I believe that most of the time the hash for a rich header is usually just an md5 of the bytes in the rich header, correct? In which case `pe.rich_header.hash.md5` would make sense to me.